### PR TITLE
[#22] [iOS] [Integrate] As a user, I can see Survey header

### LINKED ISSUE #22: [iOS] [Integrate] As a user, I can see Survey header
## Why

Home banner shows small detail of the current user.

## Acceptance Criteria

Call endpoint:
GET `/api/v1/me`

Get today information to show on the home header.

[Postman - Nimble Survey API](https://documenter.getpostman.com/view/11835486/T1LJjTqj#d85908f5-b07b-4ae7-9e71-f58c55f05f01)

### PR DESCRIPTION
**Pull request by blyscuit**
- close #22 

## What happened

Show Survey header with data from API.
 
## Insight

- Add `SurveySelectionViewModel`.
- Call api `/me` in `SurveySelectionViewModel`.
- `SurveySelectionViewModel` always return `SurveyHeaderUiModel` regardless of API error.
 
## Proof Of Work

https://user-images.githubusercontent.com/6356137/207842252-b840c5e8-44f4-4921-8149-7981cc1f04fb.mov

### CODE REVIEW

File: iosApp/SurveyTests/Sources/Specs/Presentation/Modules/SurveySelection/SurveySelectionViewDataSourceSpec.swift [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+                    getCurrentDateUseCase: getCurrentDateUseCase,
+                    getProfileUseCase: getProfileUseCase,
+                    dateTimeFormatter: DateTimeFormatterImpl()
+                )
+                dataSource = .init(
+                    viewModel: surveySelectionViewModel
+                )
+            }
+
+            describe("its init") {
```
Comment (minhnimble) [inline iosApp/SurveyTests/Sources/Specs/Presentation/Modules/SurveySelection/SurveySelectionViewDataSourceSpec.swift:39]: Shall we apply the agreed [convention](https://github.com/nimblehq/compass/discussions/1048) for describe function? 🤔 And maybe the rest of places too? 🙏
```suggestion
            describe("its init() call") {
```

File: shared/src/commonTest/kotlin/co/nimblehq/blisskmmic/presentation/modules/surveyselection/SurveySelectionViewModelTest.kt [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+    }
+
+    @AfterTest
+    fun tearDown() {
+        Dispatchers.resetMain()
+        mainThreadSurrogate.close()
+    }
+
+    @Test
+    fun `When calling fetch with success date and success user- it changes viewState with correct item`() = runTest {
```
Comment (minhnimble) [inline shared/src/commonTest/kotlin/co/nimblehq/blisskmmic/presentation/modules/surveyselection/SurveySelectionViewModelTest.kt]: Similarly with this [comment](https://github.com/blyscuit/kmm-swiftUI-IC/pull/93#discussion_r1058035769), and we can apply for all the remaining places too 🙏
```suggestion
    fun `When calling fetch with success date and success user - it changes viewState with correct item`() = runTest {
```

**Review by minhnimble — COMMENTED** [review]
I also see that you modified the Android code in this PR too, implementation-wise there is not an issue from me. At the same time, I think it is better to have a P.o.W on the changed places for Android too 🙏

File: iosApp/SurveyTests/Sources/Specs/Presentation/Modules/SurveySelection/SurveySelectionViewDataSourceSpec.swift (hunk 1 above)
Comment (blyscuit) [reply iosApp/SurveyTests/Sources/Specs/Presentation/Modules/SurveySelection/SurveySelectionViewDataSourceSpec.swift:39]: I don't think we agreed on that yet, with () it doesn't like a human readable sentence 😆, without () it looks more similar to KMM side.

**Comment by blyscuit** [comment]
> modified the Android code in this PR too, implementation-wise there is not an issue from me. At the same time, I think it is better to have a P.o.W on the changed places for Android too

@minhnimble I don't think there's an Android code change in this PR? Only on KMM module for ViewModel. They should be no Android code change as I haven't start the Android app yet 🤔

**Comment by minhnimble** [comment]
> @minhnimble I don't think there's an Android code change in this PR? Only on KMM module for ViewModel. They should be no Android code change as I haven't start the Android app yet 🤔

Oh you are right, I saw the creation of the `SurveyHeaderUiModel` but there is not a visible implementation for the Android app yet, please ignore then 🙏

**Review by minhnimble — APPROVED** [review]

**Review by doannimble — APPROVED** [review]
LGTM 🚀

**Pull request merged**